### PR TITLE
admin: include EC volumes in bucket size reporting

### PR DESCRIPTION
**Pull request by chrislusf**
## Summary
- The Object Store buckets page (`/object-store/buckets`) was computing per-collection size by iterating only regular volumes in `collectCollectionStats`, silently dropping EC-encoded data from the reported bucket size. This made usage-based billing from the admin portal unreliable.
- Walk `EcShardInfos` alongside `VolumeInfos`: add raw shard bytes to `PhysicalSize`, and the parity-stripped value (`shardBytes * DataShardsCount / TotalShardsCount`) to `LogicalSize`, matching the normalization already used by `weed shell` `cluster.status` (`weed/shell/command_cluster_status.go:437`).
- Per-collection accumulation across nodes is correct because the EC normalization ratio is constant, so summing raw shard bytes first and scaling once is equivalent to scaling per-volume.

## Test plan
- [ ] Build: `go build ./weed/admin/...`
- [ ] On a cluster with both regular and EC volumes, open `/object-store/buckets` and verify logical + physical sizes now include EC-encoded buckets
- [ ] Cross-check totals against `weed shell` → `cluster.status` for the same collection

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed collection statistics in the admin dashboard to properly account for erasure-coded volumes in addition to replicated volumes, ensuring complete and accurate storage usage metrics.

* **Tests**
  * Added comprehensive unit tests for collection statistics calculation with erasure-coded volumes, including edge cases such as uneven shard sizes and missing collection names.

### CODE REVIEW

File: weed/admin/dash/admin_server.go [hunk 1]
```diff
@@ -1698,6 +1699,20 @@ func collectCollectionStats(topologyInfo *master_pb.TopologyInfo) map[string]col
...
+						}
+						var shardBytes int64
+						for _, ss := range ecShardInfo.ShardSizes {
+							shardBytes += ss
+						}
+						data := collectionMap[collection]
+						data.PhysicalSize += shardBytes
+						data.LogicalSize += shardBytes * int64(erasure_coding.DataShardsCount) / int64(erasure_coding.TotalShardsCount)
+						collectionMap[collection] = data
+					}
```
[the remark this answers is not in the document]
Comment (chrislusf) [reply weed/admin/dash/admin_server.go:1712]: Agreed — `VolumeEcShardInformationMessage` only carries id/collection/ec_index_bits/disk_type/disk_id/shard_sizes, so there's no object count to read here. Threading one through would need: (1) a new field on the proto, (2) volume servers computing it from EC volume metadata on heartbeat, and (3) aggregation here. That's a separate change; I'll keep this PR scoped to the size fix (the immediate billing blocker) and open a follow-up issue for the count.

File: weed/admin/dash/admin_server.go [hunk 2]
```diff
@@ -1698,6 +1699,20 @@ func collectCollectionStats(topologyInfo *master_pb.TopologyInfo) map[string]col
...
+						if collection == "" {
+							collection = "default"
+						}
+						var shardBytes int64
+						for _, ss := range ecShardInfo.ShardSizes {
+							shardBytes += ss
+						}
+						data := collectionMap[collection]
+						data.PhysicalSize += shardBytes
+						data.LogicalSize += shardBytes * int64(erasure_coding.DataShardsCount) / int64(erasure_coding.TotalShardsCount)
```
[the remark this answers is not in the document]
Comment (chrislusf) [reply weed/admin/dash/admin_server.go]: Addressed in d3f9699 — the admin code no longer references `DataShardsCount`/`TotalShardsCount` directly. It now calls `erasure_coding.ShardsInfoFromVolumeEcShardInformationMessage(...).MinusParityShards().TotalSize()`, which walks the actual shard bitmap and sums data-shard bytes directly (also exact when shard sizes differ, e.g. last shard).

You're right that non-default EC policies are still a limitation, but it now lives in the `erasure_coding` helper rather than the admin layer — the correct place to fix it once per-volume shard counts are surfaced in the proto.
